Baro bias estimation, compass quick cal and range finder kinematic consistency check (#1946)
PX4/PX4-user_guide: en/config/compass.md

You will need to calibrate your compass on first use, and you may need to recalibrate it if the vehicles is ever exposed to a very strong magnetic field, or if it is used in an area with abnormal magnetic characteristics.

Two types of compass calibration are available:
1. [Complete](#complete-calibration): This calibration compensates for hard and soft iron effects by estimating an offset and a scale factor for each axis. The complete calibration is required after installing the autopilot on an airframe for the first time or when the configuration of the vehicle changed significantly.

> 1. [Partial](#partial-quick-calibration): Also known as "Quick calibration", this type of calibration only estimates the offsets to compensate for a hard iron effect. The quick calibration can be performed as a routine when preparing the vehicle for a flight, after changing the payload or simply when the compass rose seems inaccurate.

Reviewer: FYI, To many users knowing when you do these things is more important than what they do. So I've inverted the order in the description

## Performing the Calibration 

> ### Complete calibration

Reviewer: ```suggestion
### Complete Calibration
```

   
Once you've calibrated the vehicle in all the positions *QGroundControl* will display *Calibration complete* (all orientation images will be displayed in green and the progress bar will fill completely). You can then proceed to the next sensor. 

> ### Partial "Quick" calibration

Reviewer: ```suggestion
### Partial "Quick" Calibration
```

> This calibration is similar to the well-known figure-8 compass calibration done on a smartphone.
> 
> 1. Hold the vehicle in front of you and randomly perform partial rotations on all its axes.
> 1. Wait for the heading estimate to stabilize and verify that the compass rose is now pointing to the correct direction (this can take a couple of seconds).
> 
> Notes:
> - There is no start/stop for this type of calibration, the algorithm is running all the time when the vehicle is disarmed.
> - The calibration is immediately applied to the data (no reboot is required) but is saved to the calibration parameters after disarming the vehicle only (the calibration is lost if no arming/disarming sequence is performed between calibration and shutdown).
> - The amplitude and the speed of the partial rotations done in step 1 can affect the calibration quality. However, 2-3 oscillations of ~30 degrees in every direction is usually enough.

Reviewer: ```suggestion
This calibration is similar to the well-known figure-8 compass calibration done on a smartphone:

1. Hold the vehicle in front of you and randomly perform partial rotations on all its axes.
1. Wait for the heading estimate to stabilize and verify that the compass rose is pointing to the correct direction (this can take a couple of seconds).

Notes:

- There is no start/stop for this type of calibration (the algorithm runs continuously when the vehicle is disarmed).
- The calibration is immediately applied to the data (no reboot is required) but is saved to the calibration parameters after disarming the vehicle only (the calibration is lost if no arming/disarming sequence is performed between calibration and shutdown).
- The amplitude and the speed of the partial rotations done in step 1 can affect the calibration quality.
  However, 2-3 oscillations of ~30 degrees in every direction is usually enough.
```
